File: csharp/1576.replace-all-s-to-avoid-consecutive-repeating-characters.cs
/*
 * @lc app=leetcode id=1576 lang=csharp
 *
 * [1576] Replace All ?'s to Avoid Consecutive Repeating Characters
 */

// @lc code=start
public class Solution {
    public string ModifyString(string s) {
        int n = s.Length;
        var ret = s.ToCharArray();
        for (int i = 0; i < n; ++i) {
            if (s[i] != '?') {
                continue;
            }
            int pc = i == 0 ? '*' : ret[i - 1], nc = i == n - 1 ? '*' : ret[i + 1];
            for (char c = 'a'; c <= 'z'; ++c) {
                if (c != pc && c != nc) {
                    ret[i] = c;
                    break;
                }
            }
        }
        return new string(ret);
    }
}
// @lc code=end

File: java/1576.replace-all-s-to-avoid-consecutive-repeating-characters.java
/*
 * @lc app=leetcode id=1576 lang=java
 *
 * [1576] Replace All ?'s to Avoid Consecutive Repeating Characters
 */

// @lc code=start
class Solution {
    public String modifyString(String s) {
        int n = s.length();
        StringBuilder sb = new StringBuilder(n);
        char prev = '-';
        for (int i = 0; i < n; ++i) {
            char curr = s.charAt(i);
            if (curr != '?') {
                prev = curr;
                sb.append(curr);
                continue;
            }
            char next = i == n - 1 ? '-' : s.charAt(i + 1);
            if (prev != 'a' && next != 'a') {
                prev = 'a';
            } else if (prev != 'b' && next != 'b') {
                prev = 'b';
            } else {
                prev = 'c';
            }
            sb.append(prev);
        }
        return sb.toString();
    }
}
// @lc code=end

